WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ModalOverlay.swift

```diff
@@ -0,0 +1,100 @@
+import SwiftUI
+
+struct ModalOverlay<OverlayContent: View>: View {
+    @Binding var isPresented: Bool
+    @ViewBuilder let content: () -> OverlayContent
+    let onDismiss: (() -> Void)?
+
+    /// We use an internal copy of the `isPresented` state so that we can detect changes, and wrap them in a `withAnimation` call.
+    /// Without this, the fade and slide animations do not work.
+    @State private var internalIsPresented: Bool
+
+    init(isPresented: Binding<Bool>, onDismiss: (() -> Void)? = nil, content: @escaping () -> OverlayContent) {
+        self.content = content
+        self._isPresented = isPresented
+        self.onDismiss = onDismiss
+        self.internalIsPresented = isPresented.wrappedValue
+    }
+
+    var body: some View {
+        ZStack {
+            if internalIsPresented {
+                Color.black.opacity(0.5)
+                    .edgesIgnoringSafeArea(.all)
+                    .onTapGesture {
+                        dismiss()
+                    }
+                    .transition(.opacity)
+                    .animation(.easeInOut(duration: 0.1), value: internalIsPresented)
+
+                GeometryReader { geometry in
+                    VStack {
+                        content()
+                            .padding(16)
+                            .frame(width: geometry.size.width * 0.75)
+                            .frame(maxHeight: geometry.size.height * 0.8)
+                            .fixedSize(horizontal: false, vertical: true) // these three modifiers define the container size
+                            .background(Color(.tertiarySystemBackground))
+                            .cornerRadius(10)
+                            .shadow(radius: 10)
+                    }
+                    .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .center) // Ensures the container is centred
+                }
+                .transition(.move(edge: .bottom))
+                .animation(.easeInOut(duration: 0.25), value: internalIsPresented)
```

Comment:
Should we move the numbers to a Constants extension?

Reply:
This is one of those places where I feel like it would make the code less readable... and there's no (real) repetition, because the two 10s wouldn't be the same constant anyway.